scripts/stage_layer.py

```diff
@@ -0,0 +1,102 @@
+from argparse import ArgumentParser
+import logging
+from pathlib import Path
+import shutil
+import subprocess
+import sys
+from zipfile import (
+    ZipFile,
+    ZipInfo,
+)
+
+from boltons.cacheutils import cachedproperty
+import boto3
+
+from azul import (
+    config,
+)
+from azul.files import file_sha1
+from azul.logging import configure_script_logging
+
+log = logging.getLogger(__name__)
+
+
+class LayerBuilder:
+    layer_dir = Path(config.project_root) / 'lambdas' / 'layer'
+    out_dir = layer_dir / '.chalice' / 'terraform'
+
+    def __init__(self):
+        self.s3 = boto3.client('s3')
+
+    def update_layer_if_necessary(self):
+        log.info('Updating layer in bucket: %s with key %s', config.layer_bucket, self.object_key)
+        if self.update_required():
+            log.info('Updating layer.zip')
+            self.update_layer()
+        else:
+            log.info('No update necessary')
+
+    def update_required(self) -> bool:
+        log.info('Checking if layer.zip needs to be updated')
+        try:
+            # Since the object is content-addressed, just checking for the
+            # object's presence is sufficient
+            self.s3.head_object(Bucket=config.layer_bucket, Key=self.object_key)
+        except self.s3.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == '404':
+                return True
+            else:
+                raise
+        else:
+            return False
+
+    def update_layer(self, force: bool = False):
+        input_zip = self.out_dir / 'deployment.zip'
+        layer_zip = self.out_dir / 'layer.zip'
+        if force:
+            log.info('Tainting current lambda layer resource to force update')
+            command = ['terraform', 'taint', 'aws_lambda_layer_version.dependencies']
+            subprocess.run(command, cwd=Path(config.project_root) / 'terraform')
```

Comment:
Check exit code or document why its not checked, say, because the error is expected. If the latter, we may want to log a info message saying that the error is benign.

Reply:
An error will happen if you run `--force` but nothing is deployed yet, which is a conceivable (but unlikely) use case. I'll add a log message.